### ISSUE TITLE
attribute_value column in ospos_attribute_values too small

### ISSUE THREAD

**Issue by objecttothis** [opensourcepos/opensourcepos#2245]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 3.3.0
- OSPOS git commit hash is: 
- PHP version is: 5.6.38
- MySQL or MariaDB version is: 5.7.24
- OS and version is: FreeBSD 11.2
- WebServer is: Apache 2.4
- Selected language is: English

### Issue
Currently attribute_value in the ospos_attribute_values table is defined as a VARCHAR(45) which is too small for some of the custom fields.  For example, we have a custom field for author of a book.  Some text books have four or five authors which goes beyond the 45 set in the column definition.  Is there any reason why we can't make the varchar definition 255 instead of 45?  If I understand the varchar implementation in MySQL correctly then it should not be a space or speed hit on the database since varchars expand only when needed.

`CREATE TABLE IF NOT EXISTS 'ospos_attribute_values' (`
`'attribute_id' INT NOT NULL AUTO_INCREMENT,`
`'attribute_value' VARCHAR(45) NULL,`
`'attribute_datetime' DATETIME NULL,`
`PRIMARY KEY ('attribute_id')`
`) ENGINE = InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci;`

I could lump this in with my changes on #2230, but want to make sure I'm not missing something.

**Comment by SteveIreland** [comment]
I agree.  Those users that have data in the custom fields up to 255 characters might not notice that the data is being truncated during migration. This could possibly make a few users unhappy.

**Comment by objecttothis** [comment]
> 
> 
> I agree. Those users that have data in the custom fields up to 255 characters might not notice that the data is being truncated during migration. This could possibly make a few users unhappy.

I would not have known that it happened if I didn't try to run the script myself in phpmyadmin.  On that note I opened an issue #2247.  We likely need to be doing some amount of reporting during migration scripts.
